### PR TITLE
improve updateDebugString

### PR DESCRIPTION
**Pull request by CJKkkk-315**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #17957

## What this PR does / why we need it:
reduce the frequency of calls to UpdateDebugString

### CODE REVIEW

File: pkg/frontend/routine.go [hunk 1]
```diff
@@ -266,7 +266,6 @@ func (rt *Routine) handleRequest(req *Request) error {
 	//all offspring related to the request inherit the txnCtx
 	cancelRequestCtx, cancelRequestFunc := context.WithTimeout(ses.GetTxnHandler().GetTxnCtx(), parameters.SessionTimeout.Duration)
 	rt.setCancelRequestFunc(cancelRequestFunc)
-	ses.UpdateDebugString()
```
Comment (daviszhen) [inline pkg/frontend/routine.go:269]: 在func (rm *RoutineManager) Created(rs *Conn)后面，调用下UpdateDebugString

Comment (daviszhen) [reply pkg/frontend/routine.go:269]: 这个debugstr本身还是要的。只是不用每次都update.

Comment (daviszhen) [reply pkg/frontend/routine.go:269]: 然后在用到debugstr的地方。再增加个zap.String("role",...),zap.String("roleId",...)之类的。
```
zap.String("session_info", ses.debugStr)
```

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/frontend/back_exec.go [hunk 2]
```diff
@@ -915,6 +915,9 @@ func (backSes *backSession) log(ctx context.Context, level zapcore.Level, msg st
 	logger := backSes.getMOLogger()
 	if logger.Enabled(level) {
 		fields = append(fields, zap.String("session_info", backSes.GetDebugString()), zap.Bool("background", true))
+		if backSes.tenant != nil {
```
Comment (daviszhen) [inline pkg/frontend/back_exec.go:918]: 这地方不用加了。后台tenat一定是nil.

newBackSession里面，初始化backses就置为nil了

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/frontend/back_exec.go (hunk 2 above)
Comment (CJKkkk-315) [reply pkg/frontend/back_exec.go]: 好的 已修改

**Review by daviszhen — APPROVED** [review]

**Pull request merged**